ledger/ledgercore/statedelta.go

```diff
@@ -360,21 +442,14 @@ func (ad AccountDeltas) ModifiedAccounts() []basics.Address {
 
 // MergeAccounts applies other accounts into this StateDelta accounts
 func (ad *AccountDeltas) MergeAccounts(other AccountDeltas) {
-	for new := range other.Accts {
-		addr := other.Accts[new].Addr
-		acct := other.Accts[new].AccountData
-		ad.Upsert(addr, acct)
+	for _, balanceRecord := range other.Accts {
+		ad.Upsert(balanceRecord.Addr, balanceRecord.AccountData)
 	}
-
-	for aapp, idx := range other.appResourcesCache {
-		params := other.AppResources[idx].Params
-		state := other.AppResources[idx].State
-		ad.UpsertAppResource(aapp.Address, aapp.App, params, state)
+	for _, appResource := range other.AppResources {
```

Comment:
I'm trying to think through the change of iterating the AppResources themselves, instead of the cache, but don't remember what going on well enough to think it through.  Why did you change it?

It does seem more straightforward. I guess it would upsert more, if there were things in the AppResources that are not in the cache. Can that be?

Reply:
I'm going to resolve, as this looks faster and deterministic, which is certainly nice. I guess I got nervous because it seemed like a conscious choice to go through the caches. But I can't see any reason for it.